### PR TITLE
Add example how to use custom HTTP methods

### LINKED ISSUE #871: Support for HTTP extension methods
**Is your feature request related to a problem? Please describe.**
I'm trying to implement WebDAV handlers using warp, which requires filters for non-standard HTTP methods, as documented in RFC 4918. While `http::Method` allows defining custom methods, warp only provides a pre-defined set of filters for the standard methods and keeps `method_is` in `filters::method` private, so I couldn't figure out any other way to filter for custom methods.
  
**Describe the solution you'd like**
Making `method_is` public would probably be enough already. warp could provide some sugar for `Method::from_bytes`, but that is not absolutely necessary.  

**Describe alternatives you've considered**
I could't figure out any other way to filter for the method in warp.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by e-user**
Fixes #871.

This has been created as a result out of #874.

### CODE REVIEW

File: examples/custom_methods.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+use std::str::FromStr;
+
+use warp::hyper::StatusCode;
+use warp::{hyper::Method, reject, Filter, Rejection, Reply};
+
+#[derive(Debug)]
+struct MethodError;
+impl reject::Reject for MethodError {}
+
+fn method(name: &str) -> impl Filter<Extract = (), Error = Rejection> + Clone {
```
Comment (seanmonstar) [inline examples/custom_methods.rs:12]: If we assume most methods are going to be known at compile time, we could make this a `&'static str`, and then you don't need to parse it as a `Method`, or clone it, and can just compare `if m == name`, and be done with it.

Comment (e-user) [reply examples/custom_methods.rs]: Really good point. Addressed in the new commit.

File: examples/custom_methods.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+struct MethodError;
+impl reject::Reject for MethodError {}
+
+const FOO_METHOD: &'static str = "FOO";
+const BAR_METHOD: &'static str = "BAR";
+
+fn method(name: &'static str) -> impl Filter<Extract = (), Error = Rejection> + Clone {
+    warp::method()
+        .and_then(move |m: Method| {
+            async move {
```
Comment (seanmonstar) [inline examples/custom_methods.rs:18]: Looks like rustfmt is suggesting to move the `async move {` up to be on the same line as the start of the closure.

Comment (e-user) [reply examples/custom_methods.rs]: Interesting, it seems like a missing `Cargo.toml` made my rustfmt fail in applying the correct formatting. Fixed by running it with the corresponding option manually.

**Comment by e-user** [comment]
@seanmonstar Would you mind starting another workflow, please?

**Pull request merged**